Makefile

```diff
@@ -1,20 +1,86 @@
-gotags = "sqlite_vtable,sqlite_json1,static,system_libgit2"
+.PHONY: clean vet test lint lint-ci test-cover bench
+
+# default task invoked while running make
+all: clean .build/libaskgit.so .build/askgit
+
+# pass these flags to linker to suppress missing symbol errors in intermediate artifacts
+export CGO_CFLAGS = -DUSE_LIBSQLITE3
+export CPATH = ${PWD}/pkg/sqlite
+export CGO_LDFLAGS = -Wl,--unresolved-symbols=ignore-in-object-files
+ifeq ($(shell uname -s),Darwin)
+	export CGO_LDFLAGS = -Wl,-undefined,dynamic_lookup
+endif
+
+# target to build a dynamic extension that can be loaded at runtime
+.build/libaskgit.so: $(shell find . -type f -name '*.go' -o -name '*.c')
+	$(call log, $(CYAN), "building $@")
+	@go build -buildmode=c-shared -o $@ -tags="system_libgit2,shared" shared.go
+	$(call log, $(GREEN), "built $@")
+
+# target to compile askgit executable
+.build/askgit: $(shell find . -type f -name '*.go' -o -name '*.c')
+	$(call log, $(CYAN), "building $@")
+	@go build -o $@ -tags="sqlite_vtable,vtable,sqlite_json1,static,system_libgit2" askgit.go
+	$(call log, $(GREEN), "built $@")
+
+# target to download latest sqlite3 amalgamation code
+pkg/sqlite/sqlite3.c:
+	$(call log, $(CYAN), "downloading sqlite3 amalgamation source v3.35.0")
+	$(eval SQLITE_DOWNLOAD_DIR = $(shell mktemp -d))
+	@curl -sSLo $(SQLITE_DOWNLOAD_DIR)/sqlite3.zip https://www.sqlite.org/2021/sqlite-amalgamation-3350000.zip
+	$(call log, $(GREEN), "downloaded sqlite3 amalgamation source v3.35.0")
+	$(call log, $(CYAN), "unzipping to $(SQLITE_DOWNLOAD_DIR)")
+	@(cd $(SQLITE_DOWNLOAD_DIR) && unzip sqlite3.zip > /dev/null)
+	@-rm $(SQLITE_DOWNLOAD_DIR)/sqlite-amalgamation-3350000/shell.c
+	$(call log, $(CYAN), "moving to pkg/sqlite")
+	@mv $(SQLITE_DOWNLOAD_DIR)/sqlite-amalgamation-3350000/* pkg/sqlite
+
+clean:
+	$(call log, $(YELLOW), "nuking .build/")
+	@-rm -rf .build/
+
+# ========================================
+# target for common golang tasks
+
+# go build tags used by test, vet and more
+TAGS = "sqlite_vtable,vtable,sqlite_json1,static,system_libgit2"
 
 vet:
-	go vet -v -tags=$(gotags) ./...
+	go vet -v -tags=$(TAGS) ./...
 
 build:
-	go build -v -tags=$(gotags) askgit.go
+	go build -v -tags=$(TAGS) askgit.go
```

Comment:
this build step may be slightly redundant now with the above, but that's fine. It's used by this https://github.com/augmentable-dev/homebrew-askgit/blob/master/askgit.rb#L13 - just an FYI

Reply:
actually, we should drop this target and update `homebrew-askgit` to use `make .build/askgit` instead as `.build/askgit` contains the appropriate dependencies